src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,167 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size = 0;
+    private Node first;
+    private Node last;
+
     @Override
     public void add(T value) {
+        Node newNode = new Node(null, value,null);
+        if (size == 0) {
+            first = newNode;
+            last = newNode;
+        } else {
+            last.next = newNode;
+            newNode.prev = last;
+            last = newNode;
+
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        Node newNode = new Node(null, value,null);
+        if (size == index) {
+            add((T) newNode.item);
+        } else {
+            Node shiftableNode = nodeFromIndex(index);
+            if (shiftableNode.prev == null) {
+                newNode.next = shiftableNode;
+                shiftableNode.prev = newNode;
+                first = newNode;
+                size++;
+            } else {
+                shiftableNode.prev.next = newNode;
+                newNode.prev = shiftableNode.prev;
+                shiftableNode.prev = newNode;
+                newNode.next = shiftableNode;
+                size++;
+            }
+        }
+
     }
 
     @Override
     public void addAll(List<T> list) {
+        if (list != null) {
+            for (int i = 0; i < list.size(); i++) {
+                add(list.get(i));
+            }
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+
+        return nodeFromIndex(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        Node resultNode = nodeFromIndex(index);
+        T result = (T) resultNode.item;
+        resultNode.item = value;
+        return result;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        if (isEmpty() || index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException(" index error ");
+        } else {
+            Node removNode = nodeFromIndex(index);
+            T result = (T) removNode.item;
+            unlink(removNode);
+            size--;
+            return result;
+        }
     }
 
     @Override
     public boolean remove(T object) {
+
+        Node<T> desired;
+        for (desired = first; desired != null; desired = desired.next) {
+            if (desired.item == null && object == null || desired.item.equals(object)) {
+                unlink(desired);
+                size--;
+                return true;
+            }
+        }
+
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private boolean firstOrLast(int index, int size) {
+        return index < size / 2;
+    }
+
+    private Node<T> nodeFromIndex(int index) {
```

Comment:
```suggestion
    private Node<T>  getNodeByIndex(int index) {
```

названия методов должны начинаться с глаголов https://mate-academy.github.io/style-guides/java/java.html#s5.2.3-method-names